homeassistant/components/starlink/sensor.py

```diff
@@ -0,0 +1,205 @@
+"""Contains sensors exposed by the Starlink integration."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorStateClass
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DATA_RATE_BITS_PER_SECOND,
+    DATA_RATE_MEGABITS_PER_SECOND,
+    DEGREE,
+    TIME_MILLISECONDS,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONFIG_LINK_SPEED_FORMAT, DOMAIN
+from .coordinator import StarlinkUpdateCoordinator
+from .entity import StarlinkSensorEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up all sensors for this entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    config_name = entry.title
+    config_speed_format = entry.data[CONFIG_LINK_SPEED_FORMAT]
+
+    async_add_entities(
+        [
+            PingSensor(coordinator=coordinator, config_name=config_name),
+            AzimuthSensor(coordinator=coordinator, config_name=config_name),
+            ElevationSensor(coordinator=coordinator, config_name=config_name),
+            UplinkThroughputSensor(
+                coordinator=coordinator,
+                config_name=config_name,
+                data_unit=config_speed_format,
+            ),
+            DownlinkThroughputSensor(
+                coordinator=coordinator,
+                config_name=config_name,
+                data_unit=config_speed_format,
+            ),
+            LastBootTime(coordinator=coordinator, config_name=config_name),
+        ]
+    )
+
+
+class PingSensor(StarlinkSensorEntity):
```

Comment:
Did you consider using `SensorEntityDescription` pattern for these sensors? Would probably avoid creating a class for each sensor.

Reply:
I didn't even know it existed 😆
I'll look into it

Reply:
I had a quick look at SensorEntityDescription, and I can't see a method of telling the entity how to load data, so I'm not sure I can get rid of all the classes. Am I missing something?

Reply:
You can extend it with custom fields - for example, add a getter for state. See example here: https://github.com/home-assistant/core/blob/dev/homeassistant/components/wled/sensor.py#L34-L47

Reply:
I have made changes, please let me know what you think :)